airflow/contrib/hooks/spark_sql_hook.py

```
@@ -0,0 +1,150 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from airflow.hooks.base_hook import BaseHook
+from airflow.exceptions import AirflowException
+
+import logging
+import subprocess
+
+log = logging.getLogger(__name__)
+
+
+class SparkSqlHook(BaseHook):
+    """
+    This hook is a wrapper around the spark-sql binary. It requires that the 
+    "spark-sql" binary is in the PATH.
+    :param sql: The SQL query to execute
+    :type sql: str
+    :param conf: arbitrary Spark configuration property
+    :type conf: str (format: PROP=VALUE)
+    :param conn_id: connection_id string 
+    :type conn_id: str
+    :param executor_cores: Number of cores per executor
+    :type executor_cores: int
+    :param executor_memory: Memory per executor (e.g. 1000M, 2G) (Default: 1G)
+    :type executor_memory: str
+    :param keytab: Full path to the file that contains the keytab.
+    :type keytab: str
+    :param master: spark://host:port, mesos://host:port, yarn, or local
+    :type master: str
+    :param name: Name of the job.
+    :type name: str
+    :param num_executors: Number of executors to launch
+    :type num_executors: int
+    :param verbose: Whether to pass the verbose flag to spark-sql
+    :type verbose: bool
+    :param yarn_queue: The YARN queue to submit to (Default: "default")
+    :type yarn_queue: str
+    """
+    def __init__(self, 
+                sql,
+                conf=None,
+                conn_id='spark_sql_default', 
+                executor_cores=None,
+                executor_memory=None,
+                keytab=None,
+                master='yarn',
+                name='default-name',
+                num_executors=None,
+                verbose=True,
+                yarn_queue='default'
+                ):
+        self.sql = sql
```

Why would this property in particular not be visible to the outer world? From what I have seen in other hooks, this pattern isn't used. Would the `sql` property be the only one that should have private properties notation? (the reason I didn't supply a default was that without an sql query, I think the hook should throw an error, as it is the whole point of this hook to execute sql)